### DISCUSSION TITLE
Basic Usage Questions

### DISCUSSION THREAD

**Discussion by nathanielmhld** [AcademySoftwareFoundation/OpenTimelineIO#877] [Q&A] [answered]
Hello all! 
I appreciate this repository so much! I was about to set myself to developing something similar before I stumbled across it. I've been reading through the documentation, and I'm having a lot of trouble figuring out my very basic use case. I'm not sure if it's exactly what the repo is for, but I wanted to ask and see if any of you kind people could help me out.
I'd like to start in python. 
Create a new, empty timeline (I figured that one out myself)
Add clips from a directory to my timeline (just stack them one after the other, with their time/frame information automatically read in)
Add several audio tracks
Move clips and audio around a bit
Export the timeline to fcpxml or some other xml format to be read by an editing software. 

I'm going to keep trying to figure this out myself. Either way, I think such a use case might be a useful tutorial to add to the documentation, and I'd like to help with that!

**Comment by jminor** [comment, marked as answer]
Hi Nathaniel,
  I'm glad you are finding OTIO helpful. Here is some example code which may be helpful. It does something different, but you may be able to borrow pieces of it for your use case. In particular, this example code builds a timeline from scratch, and it includes an example of using ffprobe to detect the length of media files (something that OTIO doesn't handle).

https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/examples/shot_detect.py

If you're willing to write up a tutorial based on your experience, we would be happy to include that in the documentation, wiki, or something similar.

**Reply by nathanielmhld** [reply]
Hi jminor! 
Thank you for your response. I took your advice and used shot_detect as a reference. I wasn't able to get the functions there to work exactly, but after making use of the ffprobe-python library and changing a few small things, I made a lot of progress. I'm really struggling with adding audio though. I know it's possible. Turns out, final cut pro won't even load an fcpxml file if a clip has audio, but the fcpxml doesn't indicate the audio of the clip. Could you point me towards some documentation on adding audio to a clip? (Or, if that's not how it works, even adding it to a new track. I can't seem to figure out how to flag an external reference as audio when it's audio from a video file, say mp4). 

If you can help me accomplish my goal, I will happily document the process and submit it so other newbies don't run into the same issues that I have.

**Reply by apetrynet** [reply]
Hi @nathanielmhld !

At the moment there is no way to indicate if a clip is or contains audio. What you need to do is create an "audio" clip (just a regular clip really) and place it in an audio track. If it's the sound for a video clip just place it at the same location in an autio track as the corresponding video clip. 
To create an audio track you need to pass a track kind like: 
```python
audio_track = otio.schema.Track('A1', kind=otio.schema.Track.Kind.Audio)
```
You can't reuse the same clip instance in both audio and video tracks so you'll need to create two separate clips. If the audio is the same media source as the video you can use `audio_clip = video_clip.clone()`. If not, just create an ExternalReference to the audio media.

Hope this helps you further along.
